app/services/userService.js

```diff
@@ -26,25 +27,31 @@ exports.comparePassword = async (password, dbPassword) => {
 
 exports.findUserByEmail = email => UserModel.findOne({ where: { email: formatEmail(email) } });
 
-exports.createUser = async ({ firstName, lastName, email, password }) => {
+exports.createUser = async ({ firstName, lastName, email, role, password }) => {
   const hash = await this.encryptPassword(password);
 
   const user = await UserModel.create({
     firstName,
     lastName,
     email: formatEmail(email),
-    password: hash
+    password: hash,
+    role
   });
 
-  const token = generateToken(user.id);
+  const token = generateToken(user);
 
   logger.info(`Usuario creado: ${user.firstName} ${user.lastName}`);
   return { user, token };
 };
 
 exports.login = user => {
-  const token = generateToken(user.id);
+  const token = generateToken(user);
 
   logger.info(`Usuario logueado: ${user.firstName} ${user.lastName}`);
   return { user, token };
 };
+
+exports.updateUser = async (userId, body) => {
+  const userUpdated = await UserModel.update(body, { where: { id: userId }, returning: true });
+  return userUpdated[1][0].get();
```

Comment:
Will always exist the array of arrays to access `userUpdated[1][0]`?

Reply:
Yes!
From docs: "By default, .update() returns an array with two possible objects. The first is the number of rows affected by the update. It's always included. The second element is an array of the data instances from the rows themselves. You must be using Postgres and have set returning: true for this second data to return."